homeassistant/components/webostv/__init__.py

```diff
@@ -52,6 +52,7 @@
                         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
                         vol.Optional(CONF_ON_ACTION): cv.SCRIPT_SCHEMA,
                         vol.Optional(CONF_ICON): cv.string,
+                        vol.Optional(CONF_PLEX_ENTITY): cv.string,
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration or changes.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
I have started to work on such PR (https://github.com/home-assistant/core/pull/53256), but got stuck waiting for changes upstream, still waiting for them

Reply:
@thecode could you link to the upstream changes that are holding back config flow for the webostv integration?
Would love to see the webostv have config flow and had the feeling you were really close.

Reply:
I'm confused now, https://github.com/home-assistant/core/pull/53256 is merged, but there is a TODO-list in its description?

Reply:
@emontnemery it was merged to home-assistant:thecode-add-webostv-config-flow not the dev branch since it was apperently not fully ready yet (I think).
I was following that PR, so hopfully it will be revived now.

Reply:
The two items in the todo list awaits changes upstream.
1. This PR: https://github.com/bendavid/aiopylgtv/pull/31 which disable the local storage file used by the library, there are some ways to workaround this, but few months ago I was promised they will happen soon so I decided to wait for this change.
2. To solve the unique id when migrating to Zeroconf / SSDP discovery a change upstream needed to read a unique id from the TV, I already made this change locally but did not make a PR for it, since my first PR was waiting.

@emontnemery the reason there is TODO is that I initially started to work on finishing a PR which was created by someone else and during review and testing I discovered it is far from being ready (the main problem was related to the issues in the TOOD list), the plan was to wait for the changes upstream and finish the PR. since the PR was already big @MartinHjelmare suggested to finish the review and merge it to a branch.

For now I don't think the with the current progress upstream it will be possible to finish my PR, however just today I had a contact from someone who forked the library and is maintain it, the storage is already fixed there (all storage related functions are moved to a storage class). I am considering changing the library to this https://github.com/chros73/bscpylgtv and finish this PR.

Reply:
I had a call with the owner of `bscpylgtv` he will help me with the required changes, as a first step I will verify that this lib doesn't break anything with the integration and plan to resume working on the config flow PR.

Reply:
@thecode that is greath news, If you need any help or testing let me know.
I have a dev setup and a LG Web OS TV.

Reply:
This is a custom integration based on `bscpylgtv` fork:
https://github.com/thecode/home-assistant-custom-components/tree/main/webostv
I am already using it in my prod environment with no issues.
Will highly appreciate testing as it will be the base for the config flow PR.

Reply:
@thecode sorry for taking so long, but I finally got some time to test it out.
Unfortunatley it did not work, both on my dev and production env the custom component does not work.

On the dev env if I start with a clean configuration, it will ask me to couple the TV through a Notification, I press the button on the notification, the TV asks me to accept, I hit YES and then nothing happens, the notification will ask again to couple, and if I then click the button in the notification the TV does not give a popup again (probably because it is already accepted), but the notification stays and no entity is beeing created.

If I copy my `webostv.conf` file from my production env to the dev enviroment with the custom component it will create the entity, but the entity will not update and does not represent the correct state and I keep getting: 
`2022-01-08 13:59:16 WARNING (MainThread) [homeassistant.components.media_player] Updating webostv media_player took longer than the scheduled update interval 0:00:10`

On my production env if I use the custom_component (with the config from the official integration) it gives me this:
```
Logger: homeassistant.setup
Source: setup.py:231
First occurred: 14:11:55 (1 occurrences)
Last logged: 14:11:55

Setup of webostv is taking longer than 300 seconds. Startup will proceed without waiting any longer


Logger: homeassistant.setup
Source: setup.py:291
First occurred: 14:11:55 (2 occurrences)
Last logged: 14:11:55

Unable to prepare setup for platform webostv.media_player: Unable to set up component.
Unable to prepare setup for platform webostv.notify: Unable to set up component.


Logger: homeassistant.components.notify
Source: components/notify/legacy.py:52
Integration: Meldingen (documentation, issues)
First occurred: 14:11:55 (1 occurrences)
Last logged: 14:11:55

Unknown notification service specified
```
and the entity is never setup correctly.

Reply:
@thecode I have a LG OLED 55EG920v with the latest updates installed.
firmware version: 04.26.10
webos version: 2.2.3-1353 (beehive-biscayne)

Note that this TV turns off its network card when the TV is in standby meaning it can not communicate to HomeAssistant while in standby and cannot be woken by a magic packet. (I use CEC to wake the TV).

Reply:
Sorry for not notifying here, it was too long and was updated after I wrote here, the version I had initially asked to test did not work correctly when TV is off, since than I have updated to a newer `bscpylgtv` which fixed it and broke something else. Trying to fix it was complicated since the owner for `bscpylgtv` does not use home assistant or have issues enabled on his repo. We have started our own fork for the library and an initial version was released yesterday. 
I will notify when there is something ready to test.

Reply:
@thecode thanks for the update.
Maybe we schould have a better place to discuss the config flow, since this PR is not even about the config flow...
Do you already have a PR open somewhere, or could you maybe make a new issue or something where we can discuss if you need testing and where I can follow updates.

Reply:
I agree, was also thinking the same, this [branch](https://github.com/home-assistant/core/tree/thecode-add-webostv-config-flow) contains the WIP for config flow, I will probably make a PR today to update the library there, will tag you so you can follow up the changes.